src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -234,16 +266,15 @@ protected void runTool(final JavaSparkContext ctx) {
      *
      * This will be replaced by a parallel writer similar to what's done with {@link org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSink}
      */
-    private void writeVariants(JavaRDD<VariantContext> variants, HaplotypeCallerEngine hcEngine) {
+    private static void writeVariants(String outputFile, JavaRDD<VariantContext> variants, HaplotypeCallerEngine hcEngine, SAMSequenceDictionary referenceDictionary, SAMSequenceDictionary readsDictionary) {
```

Comment:
I need to investigate this further since I was hitting problems with the sequence dictionary in a whole genome BAM being different to the reference.